transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollSocketChannel.java

```diff
@@ -110,6 +114,33 @@ protected AbstractEpollUnsafe newUnsafe() {
         return new EpollSocketChannelUnsafe();
     }
 
+    @Override
+    boolean doConnect0(SocketAddress remote) throws Exception {
+        if (Native.IS_SUPPORTING_TCP_FASTOPEN && config.isTcpFastOpenConnect()) {
```

Comment:
What happens if connect() is called twice with fast open turned on?

Reply:
The behavior still seems to be maintained. SocketMultipleConnectTest is using the new permutations and seems to pass just fine

Reply:
What I mean is, it appears this method will discard the second `doConnect()`, where as the normal channel will complain.    Does EpollSocketMultipleConnectTest account for the fast open variant?

Reply:
yeah, the actual (untouched) `doConnect()` method will check if a remoteAddress is already set and fail before `doConnect0()` (which has an override here) is called, so the AlreadyConnectedException will still be thrown using the expected behavior (it doesnt rely on the actual socket state). EpollSocketMultipleConnectTest uses the new permutation set (with the client fast open) and passes fine. So thankfully that should still work like it normally does. 

The only tests that needed to exclude the new (client fast open) permutation are mostly based around behavior of the stateful connection phase, without the client sending data, which is expected. The only real unexpected one that failed is the FileRegion test, which I might able to fix... maybe (sendFile doesn't support MSG_FASTOPEN). Still playing around with options there.